samza-kafka/src/main/scala/org/apache/samza/checkpoint/kafka/KafkaCheckpointManager.scala

```diff
@@ -183,96 +152,100 @@ class KafkaCheckpointManager(
       checkpoints.put(taskName, checkpoint) // replacing any existing, older checkpoints as we go
     }
 
-    readLog(CHECKPOINT_LOG4J_ENTRY, shouldHandleEntry, handleCheckpoint)
+    readLog(shouldHandleEntry, handleCheckpoint)
     checkpoints.toMap /* of the immutable kind */
   }
 
+  private def getEarliestOffset(topic: String, partition: Partition): SystemStreamPartitionMetadata = {
+    val systemAdmin = getSystemAdmin()
+    val metaDataMap: java.util.Map[String, SystemStreamMetadata] = systemAdmin.getSystemStreamMetadata(Collections.singleton(topic))
+    val checkpointMetadata: SystemStreamMetadata = metaDataMap.get(topic)
+    if(checkpointMetadata == null)
+      throw new SamzaException("Cannot get metadata for system=%s, topic=%s" format (systemName, topic))
```

Comment:
Prefer `{}` even for one line statements to make code immune to accidental refactors later

Reply:
you are better then checkStyle! Done.